docs/registry/create/_index.md

```diff
@@ -11,154 +11,110 @@ tags:
+For example, if your organization namespace is `acme`, and you have written a new base implementation named `mybase` which you have pushed to a repository named `my-custom-base-repo`, you would use the namespace `acme:my-custom-base-repo:mybase` for your model.
 
-When implementing a custom {{< glossary_tooltip term_id="model" text="model" >}} of an existing [service](/services/), valid [APIs](/build/program/apis/) always have the following parameters:
+Determine the model name you want to use based on these requirements, then proceed to the next section.
 
-- `namespace`: `rdk`
-- `type`: `service`
-- `subtype`: any one of [these service proto files](https://github.com/viamrobotics/api/tree/main/proto/viam/service), for example `navigation`
+If you do not intend to [upload your module](/registry/upload/) to the [Viam registry](https://app.viam.com/registry), you do not need to use your organization's namespace as your model's namespace.
 
-#### Unique cases
+See [Naming your model](/registry/#naming-your-model-namespacerepo-namename) for more information.
 
-If you are using unique hardware that does not already have an [appropriate API](/build/program/apis/#component-apis) defined to support it, you can use the [generic API](/components/generic/) to add support for that unique hardware type to your machine.
+### Write your new resource model definition
 
-Some use cases may require you to define a new API, or to deploy custom components using a server on a remote part.
-For more information, see [Advanced Modular Resources](/registry/advanced/).
+Follow the instructions below to define the capabilities provided by your model, for the language you are using to write your module code:
 
-## Create a custom module
+{{% alert title="Note: Pin numbers" color="note" %}}
 
-A custom module wraps one or more {{< glossary_tooltip term_id="model" text="models" >}}.
-To create a custom module, follow these steps:
+If your module references {{< glossary_tooltip term_id="pin-number" text="pin numbers" >}}, you should use physical board pin numbers, _not_ GPIO (BCM) numbers, to maintain consistency across {{< glossary_tooltip term_id="resource" text="resources" >}} from different sources.
 
-1. [Code a new resource model](#code-a-new-resource-model) server.
-2. [Code a main program](#code-a-main-entry-point-program).
-3. [Compile or package](#prepare-the-module-for-execution) the module into a single executable.
+{{% /alert %}}
 
-### Code a new resource model
+{{< tabs >}}
+{{% tab name="Python" %}}
 
-{{% alert title="Naming your model" color="tip" %}}
+First, inspect the built-in class provided by the resource API that you are extending.
 
-Use the naming schema: `namespace:repo-name:name`.
+For example, if you want to add support for a new [base component](/components/base/) to Viam, you would start by looking at the built-in base component class.
```

Comment:
Grammar: If sentences that use conditional clauses must use past tense (simple past or past perfect) in the if-clause of the sentence